crates/infra/cli/src/toolchains/napi/cli.rs

```diff
@@ -69,7 +69,11 @@ impl NapiCli {
                 "index.js" | "index.d.ts" => {
                     source_files.push(output_dir.join(file_name));
                 }
-                file if file.ends_with(".node") && node_binary.is_none() => {
+                file if Path::new(file)
+                    .extension()
+                    .is_some_and(|ext| ext.eq_ignore_ascii_case("node"))
```

Comment:
Q: why ignoring ascii case?
An exteremely rare case, but in general, I would prefer to be more strict here, as it might fail to load on *nix systems if it had the wrong case.

Reply:
See https://rust-lang.github.io/rust-clippy/master/index.html#/case_sensitive_file_extension_comparisons. I've commented about it in https://github.com/NomicFoundation/slang/pull/695#discussion_r1415842706; Agreed that we probably won't ever hit but it's slightly more correct.

Reply:
> Noisier, but I guess slightly more correct

But in that case, we DON"T want to match `index.Node`. Right? Otherwise, this `require()` would fail:

https://github.com/NomicFoundation/slang/blob/7a3c1fbe794055c35faaeddcb9dc8814cdd5c91e/crates/solidity/outputs/npm/package/src/generated/index.js#L45

Reply:
Could you elaborate a bit?

The `Path::extension` returns the portion of the file name after the final dot, so the below is true:
```rust
assert_eq!(Path::new("index.linux-x64-gnu.node").extension(), Some(std::ffi::OsStr::new("node")));
```

Or do you mean that if we require `.node`-suffixed files then we de facto require them to be always `.node`, including the casing?

At a quick glance it seems that Node.js `require` is in fact case-sensitive on case-sensitive file systems.

Do you want me to revert this change?

Reply:
> At a quick glance it seems that Node.js require is in fact case-sensitive on case-sensitive file systems.

That is correct. I think we can still use `extension()` if you like, but compare it considering case.

Reply:
Actually, now that I think about it, napi could decide to emit `.NODE` files and correctly generate the index.js with `.NODE` requires that we would miss. However, we do match on the `index.js` considering case above anyway, so we're not trying to detect such cases as well and it's probably an easy mistake to spot/fix once we encounter this.

As such, I'm inclined to revert this change and add an allow :+1:

Reply:
Done in 2983ff0489bbb850811689c0598f630024c7ce54